solc/CommandLineInterface.cpp

```diff
@@ -1097,6 +1097,21 @@ void CommandLineInterface::outputCompilationResults()
 {
 	solAssert(m_options.input.mode == InputMode::Compiler || m_options.input.mode == InputMode::CompilerWithASTImport, "");
 
+	std::set<std::string> uncompiledContracts;
+	for (auto const& contract: m_compiler->contractNames())
+		if (objectWithLinkRefsHex(m_compiler->object(contract)).empty())
+			uncompiledContracts.insert(contract.substr(contract.find_last_of('/') + 1));
```

Comment:
Are you trying to extract the file name here? Fully-qualified names in `CompilerStack` have the form `<source unit name>:<contract name>` so you should split on the last `:` (`<contract name>` cannot contain `:`).

Also, since contract names are not necessarily unique, I think we should list the file name too.